hs_gabbs_web_app/receivers.py

```diff
@@ -0,0 +1,34 @@
+from django.dispatch import receiver
+from django.conf import settings
+
+from hs_tracking.signals import post_tracking_web_app_launch_url
+
+from .utils import push_res_to_geohub
+
+
+@receiver(post_tracking_web_app_launch_url)
+def webapp_launch_url_processing(sender, **kwargs):
+    request = kwargs['request']
+    url = kwargs['url']
+
+    if settings.GEOHUB_HOMEPAGE_URL in url:
```

Comment:
is there an app id or something we could use instead of the url?

Reply:
along the same lines, I don't like that this method is called for every app launched.  I need to look into how apps are handled to see if I can work out a pattern I like.

Reply:
what do you think about moving this method into the resource?  That way when you open an app, the openwith code could check to see if there is a preprocessing hook for each resource and only call it for resources that implement it.

BaseResource seems to have a pretty wide usage.  I wouldn't expect it to be used for a web app, does anyone know why it is?  I'm still digesting a lot of what's here and don't have a clear picture yet.

Reply:
@sblack-usu I looked into web app tool resource model and metadata implementation and don't think we have a pattern or id or anything we can identify a specific web app resource that is to be opened with. I am looping in @zhiyuli who implemented web app tool resource in case he has ideas. I don't think we can move this method into the resource, since this receiver function is only executed when the MultiSpec web app is being opened with, which is mapped to this hs_tracking view function to track web app opening event, so the hook has to be put there to only push resource to mygeohub iRODS zone when the request action is taking place. Currently we can only use either URL or the web app tool title to identify that specific web app tool resource, which has to be compared against a mygeohub specific URL or title for special processing. I totally agree with you that it'd be great if we could only invoke this receiver when needed to, and the only way to achieve this is to filter by signal sender. However, the sender in our case is that AppLaunch view class in hs_tracking app, and we cannot distinguish by that sender. I don't think this is too bad, though, since it just returns if it is not mygeohub MultiSpec app based on url check. Of course, if we could avoid that, it'd be great, but I don't see a way around this. @zhiyuli It'd be great if you could take a look at this PR as well since it is closely web app related.